src/pages/api/achievement.ts

```diff
@@ -18,6 +18,9 @@ export default async function handler(
   req: NextApiRequest,
   res: NextApiResponse<any>
 ) {
+  if (req.headers.authorization !== process.env.NEXT_API_BEARER_TOKEN) {
+    res.status(401).json({});
```

Comment:
In my opinion, in this case, returning a 403 error code is better since we don't return how to authenticate, as explained in https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401 🙂 
```suggestion
    res.status(403).json({});
```